app.js

```diff
@@ -81,6 +82,66 @@ httpsApp.get("/", function(req, res) {
   res.sendFile(__dirname + "/public/login.html");
 });
 
+// Search through JSON and add interaction information to temporary JSON, then send back to client
+httpsApp.post("/search", function(req, res) {
+  var filter = req.body.filter;
+  var searchOption = req.body.searchValue;
+  var numItems = req.body.numItems;
+  var json = JSON.parse(fs.readFileSync(file, 'utf8'));
+  var jsonData = {};
+  jsonData.interactions = []
+  var counter = 0;
+  for (var interaction in json.databaseContent) {
+    if (searchOption == "session_id") {     
+      if (json.databaseContent[interaction].sessionId.slice(23,44).toLowerCase().indexOf(filter.toLowerCase()) > -1) {
```

Comment:
constants for the slices

const SESSIONID_SLICE_START = 23
const SESSIONID_SLICE_END= 44
const USERID_SLICE_START = 18
const USERID_SLICE_END = 40